src/standardize.jl

```diff
@@ -0,0 +1,326 @@
+"""
+    μ, σ = standardize!(X[, μ, σ; obsdim, operate_on])
+
+or
+
+    μ, σ = standardize!(D[, μ, σ; operate_on])
+
+Normalize `X` along `obsdim` according to X = (X - μ) / σ.
+If μ and σ are omitted they are computed such that variables have a mean of zero
+
+
+
+`μ`         :  Vector or value describing the translation.
+               Defaults to mean(X, 2)
+
+`σ`         :  Vector or value describing the scale.
+               Defaults to std(X, 2)
+
+`obsdim`    :  Specify which axis corresponds to observations.
+               Defaults to obsdim=2 (observations are columns of matrix)
+               For DataFrames `obsdim` is obsolete and centering occurs
+               column wise.
+
+`operate_on`:  Specify the indices of columns or rows to be centered.
+               Defaults to all columns/rows.
+               For DataFrames this must be a vector of symbols, not indices
+               E.g. `operate_on`=[1,3] will perform centering on columns
+               with index 1 and 3 only (if obsdim=1, else rows 1 and 3)
+
+
+Note on DataFrames:
+Columns containing `NA` values are skipped.
+Columns containing non numeric elements are skipped.
+
+Examples:
+
+    X = rand(4, 100)
+    x = rand(10)
+    D = DataFrame(A=rand(10), B=collect(1:10), C=[string(x) for x in 1:10])
+
+    μ, σ = standardize!(X, obsdim=2)
+    μ, σ = standardize!(X, ObsDim.First())
+    μ, σ = standardize!(X, obsdim=1, operate_on=[1,3]
+    μ, σ = standardize!(X, [7.0,8.0], [1,1], obsdim=1, operate_on=[1,3]
+    μ, σ = standardize!(D)
+    μ, σ = standardize!(D, operate_on=[:A,:B])
+    μ, σ = standardize!(D, [-1,-1], [2,2], operate_on=[:A,:B])
+"""
+function standardize!(X, μ, σ; obsdim=LearnBase.default_obsdim(X), operate_on=default_scaleselection(X, convert(ObsDimension, obsdim)))
+    standardize!(X, μ, σ, convert(ObsDimension, obsdim), operate_on)
+end
+
+function standardize!{T,N}(X::AbstractArray{T,N}, μ, σ, ::ObsDim.Last, operate_on)
+    standardize!(X, μ, σ, ObsDim.Constant{N}(), operate_on)
+end
+
+function standardize!(X; obsdim=LearnBase.default_obsdim(X), operate_on=default_scaleselection(X, convert(ObsDimension, obsdim)))
+    standardize!(X, convert(ObsDimension, obsdim), operate_on)
+end
+
+function standardize!{T,N}(X::AbstractArray{T,N}, ::ObsDim.Last, operate_on)
+    standardize!(X, ObsDim.Constant{N}(), operate_on)
+end
+
+function standardize!{T,N,M}(X::AbstractArray{T,N}, obsdim::ObsDim.Constant{M}, operate_on)
+    μ = vec(mean(X, M))[operate_on]
+    σ = vec(std(X, M))[operate_on]
+    standardize!(X, μ, σ, obsdim, operate_on)
+end
+
+function standardize!{M}(X::AbstractVector, ::ObsDim.Constant{M}, operate_on)
+    μ = mean(X)
+    σ = std(X)
+    for i in operate_on 
+        X[i] = (X[i] - μ) / σ
+    end
+    μ, σ
+end
+
+function standardize!(X::AbstractMatrix, μ::AbstractVector, σ::AbstractVector, ::ObsDim.Constant{2}, operate_on)
+    σ[σ .== 0] = 1
```

Comment:
It is meant to catch the case of constant variables with std(var) = 0 and prevent the division by 0.
Could also do a `if sigma != 0 do rescale`